File: Ejemplos/Ficheros-Java/src/main/java/es/joseluisgs/binario/Binario.java
package es.joseluisgs.binario;

import java.io.*;

public class Binario {
    public static void main(String[] args) {
        leerFichero();
        escribirFichero();
    }

    private static void escribirFichero() {
        String myPath = System.getProperty("user.dir");
        String myFile = myPath + File.separator + "data" + File.separator + "fichero-texto-2.txt";
        System.out.println(myFile);
        // Existe y podemos leerlo?
        File fDestino = new File(myFile);

        try (FileOutputStream fileOutputStream = new FileOutputStream(fDestino)) {
            String texto = "Hola mundo";
            fileOutputStream.write(texto.getBytes());
        } catch (IOException e) {
            e.printStackTrace();
        }

        try (BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(new FileOutputStream(fDestino))) {
            String texto = "Hola mundo";
            bufferedOutputStream.write(texto.getBytes());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private static void leerFichero() {
        String myPath = System.getProperty("user.dir");
        String myFile = myPath + File.separator + "data" + File.separator + "fichero-texto.txt";
        System.out.println(myFile);
        // Existe y podemos leerlo?
        File fOrigen = new File(myFile);

        try (FileInputStream fileInputStream = new FileInputStream(fOrigen)) {
            int singleCharInt;
            char singleChar;

            while((singleCharInt = fileInputStream.read()) != -1) {
                singleChar = (char) singleCharInt;
                System.out.print(singleChar);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        try (BufferedInputStream bufferedInputStream = new BufferedInputStream(new FileInputStream(fOrigen))) {
            int singleCharInt;
            char singleChar;
            while((singleCharInt = bufferedInputStream.read()) != -1) {
                singleChar = (char) singleCharInt;
                System.out.print(singleChar);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Ejemplos/Ficheros-Kotlin/src/main/kotlin/binario/Binario.kt
package binario

import java.io.File

fun main() {
    leerFichero()
    escribirFichero()
}

fun escribirFichero() {
    val programPath = System.getProperty("user.dir")
    val filePath = "$programPath${File.separator}data${File.separator}fichero-texto-2.txt"
    println("Escribiendo fichero: $filePath")

    val fileDestino = File(filePath)

    // writeBytes
    fileDestino.writeBytes("Hola mundo!".toByteArray())

    // outputStream
    fileDestino.outputStream().use { it.write("Hola mundo!".toByteArray()) }

    // bufferedOutputStream
    fileDestino.outputStream().buffered().use { it.write("Hola mundo!".toByteArray()) }

    // con bufferedWriter
    fileDestino.bufferedWriter().use { it.write("Hola mundo!") }
}

fun leerFichero() {
    val programPath = System.getProperty("user.dir")
    val filePath = "$programPath${File.separator}data${File.separator}fichero-texto.txt"
    println("Leyendo fichero: $filePath")

    val fileOrigen = File(filePath)

    // readBytes
    val bytes = fileOrigen.readBytes().toString(Charsets.UTF_8)
    println(bytes)
    println()
    // con un inputStream
    fileOrigen.inputStream().use { println(it.readBytes().toString(Charsets.UTF_8)) }
    println()
    // con buffered
    fileOrigen.inputStream().buffered().use { println(it.readBytes().toString(Charsets.UTF_8)) }
}
